server/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricIT.java

```diff
@@ -1016,4 +1058,37 @@ public void testConflictingAggAndScriptParams() {
         SearchPhaseExecutionException ex = expectThrows(SearchPhaseExecutionException.class, builder::get);
         assertThat(ex.getCause().getMessage(), containsString("Parameter name \"param1\" used in both aggregation and script parameters"));
     }
+
+    public void testAggFromContext() {
```

Comment:
Added a new test here for now to get some specific coverage easily. I think what actually makes sense here is to run all the test cases here against both old param and new context variable scripts, while both are implemented.